scripts/build.sh

```diff
@@ -119,7 +119,7 @@ fi
 # https://github.com/opensearch-project/k-NN/issues/975
 GCC_REQUIRED_VERSION_CEILING=8.0.0
 COMPARE_VERSION_CEILING=`echo $GCC_REQUIRED_VERSION_CEILING $GCC_VERSION | tr ' ' '\n' | sort -V | uniq | tail -n 1`
-if [ ! "$COMPARE_VERSION_CEILING" = "$GCC_REQUIRED_VERSION_CEILING" ]; then
+if [ "$COMPARE_VERSION_CEILING" != "$GCC_REQUIRED_VERSION_CEILING" ] && (echo "$OSTYPE" | grep -i linux); then
```

Comment:
can we use "$PLATFORM" from top instead ?

Reply:
This is temp code specifically for centos7 before we move to rockylinux8.
Both the x64 and the arm64 centos7 image is built based on the same dockerfile thus have same gcc version.
As long as linux is specifically locked to version above 4.9 and lower than 8 we are good.